### ISSUE TITLE
Request implementing SOL_IP for sycall getsockopt

### ISSUE THREAD

**Issue by dequeueing** [asterinas/asterinas#1683]
### Feature Description

Hi. I hope Asterinas can support the subcommand `SOL_IP` for syscall **getsockopt**.
https://man7.org/linux/man-pages/man2/getsockopt.2.html

### Problem or Need

Asterinas does not support this subcommand for now:

```rust
pub fn new_raw_socket_option(
    level: CSocketOptionLevel,
    name: i32,
) -> Result<Box<dyn RawSocketOption>> {
    match level {
        CSocketOptionLevel::SOL_SOCKET => new_socket_option(name),
        CSocketOptionLevel::SOL_TCP => new_tcp_option(name),
        _ => todo!(),
    }
}
```

I am trying to run some applications, which use this subcommand `SOL_IP`. However, it is too diffucult to workaround in the user space, so I hope it can be supported in the kernel space?

### Suggested Implementation

I am not familiar with the network aspect of Asterinas. I would be happy if you could point out some solutions to work with it. 

### Additional Context and Screenshots

To be more specific, I am trying to run Python programs on Asterinas. Some applications require loopback network to do process communication. These pragrams are intricately encapsulated, so it is difficult to modify the implementation from inside the Python codes.

![image](https://github.com/user-attachments/assets/3d837ac5-d62a-4a44-b3b4-0dbe085dce38)
![image](https://github.com/user-attachments/assets/647f7700-22dc-4b81-937a-44dd33fe36e9)

**Comment by StevenJiang1110** [comment]
Certainly, features like `SOL_IP` should be implemented in kernel space. 

However, based on your log, it doesn’t seem that the issue is related to `SOL_IP`. The most explicit error message is "the address is not available from the local machine," which suggests a socket bind failure. The most common cause of this issue is the socket attempting to bind to 0.0.0.0, which means binding to all addresses on Linux. However, Asterinas does not support binding to 0.0.0.0; each socket must be bound to a specific address (e.g., 127.0.0.1). You might consider setting the log level to `debug` to determine which address the socket is attempting to bind to.

**Comment by dequeueing** [comment]
> Certainly, features like `SOL_IP` should be implemented in kernel space.
> 
> However, based on your log, it doesn’t seem that the issue is related to `SOL_IP`. The most explicit error message is "the address is not available from the local machine," which suggests a socket bind failure. The most common cause of this issue is the socket attempting to bind to 0.0.0.0, which means binding to all addresses on Linux. However, Asterinas does not support binding to 0.0.0.0; each socket must be bound to a specific address (e.g., 127.0.0.1). You might consider setting the log level to `debug` to determine which address the socket is attempting to bind to.

Thanks for the info. I have known this point earlier before posting this issue and therefore binding all the address to 127.0.0.1. Here is the new log: 


![image](https://github.com/user-attachments/assets/05f10b81-798c-4f8f-9063-483fbc98c2a7)


where I manually add a error msg:

![image](https://github.com/user-attachments/assets/64ed600f-8300-4eaf-b25e-a6459b41a6b9)

**Comment by StevenJiang1110** [comment]
Fixed by #1804
